### ISSUE TITLE
colexec: add support for JSONFetchText binary operator

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#49470]
Add support for `JSONFetchText` binary operator. For more details, see #49463.

**Comment by deathname** [comment]
Hi, Can I work on this issue

**Comment by yuzefovich** [comment]
That would be great! I'll assign the issue to you.

**Comment by deathname** [comment]
Thanks, @yuzefovich 

I think this issue is similar to  [https://github.com/cockroachdb/cockroach/pull/49818/files](url).
I just need to add one more switch case handling for JSONFetchText and return type.DString  in **overloads_bin.go** just as similar to the above-mentioned url or there is something else ?

**Comment by yuzefovich** [comment]
Yes, I think the solution will be very similar to #49818.

**Comment by deathname** [comment]
What should be expected output for `EXPLAIN (VEC) SELECT _json -> 2 ->> 'a' FROM many_types` as per vectorize_overloads input
`--- Node 1 -> *rowexec.tableReader` is it correct ?

**Comment by yuzefovich** [comment]
No, this means that the old row-by-row engine is used, not the vectorized engine.

I think you might be missing a few things. Please take a look at the detailed message [here](https://github.com/cockroachdb/cockroach/issues/49463#issue-623538582).

**Comment by haseth** [comment]
Hi @yuzefovich, this should be pretty simple to pick and is not blocked by #49780 I think. Can you check it once? 
JSONFetchText only requires types.String as 2nd operand.

**Comment by yuzefovich** [comment]
@haseth I think you're misunderstanding the limitation described by #49780. Currently we are not able to support a case of `binFn(datumVec, datumVec) = <native type>`, and this is exactly the case we have here - `binFn(datumVec, coldata.Bytes) = coldata.Bytes`.
